addons/html_editor/static/src/plugin.js

```diff
@@ -111,11 +111,19 @@ export class Plugin {
     addDomListener(target, eventName, fn, capture) {
         const handler = fn.bind(this);
         target.addEventListener(eventName, handler, capture);
-        this._cleanups.push(() => target.removeEventListener(eventName, handler, capture));
+        const hash = eventName + (Math.random() + 1).toString(36).substring(7);
+        this._cleanups[hash] = () => target.removeEventListener(eventName, handler, capture);
+        return hash;
+    }
+    removeDomListener(hash) {
+        if (this._cleanups[hash]) {
+            this._cleanups[hash]();
+            delete this._cleanups[hash];
+        }
```

Comment:
Have you discussed with the JS guys how they handle the "enter dilemma"?

Shouldn't you just add the event listener yourself and manage it instead of changing the `Plugin` class? 
I mean not using the `Plugin.addDomListerner` method in this case (and maybe extend the `destroy` method in your plugin to remove your listener in this case).

Reply:
didn't discuss it, but I'm open to change.
I think my current approach is ok, because it ensure my listener will be garbage collected in case of unexpected editor destroy from owl.